src/main/java/ingeniousthings/sigfox/decoder/SigfoxDecoder/SigfoxDecoder.java

```
@@ -0,0 +1,128 @@
+				break;
+			case "byte" :
+				type = VarType.BYTE;
+				break;
+			case "int" :
+				type = VarType.INTEGER;
+				break;
+			case "float" :
+				type = VarType.FLOAT;
+				break;
+			case "double" :
+				type = VarType.DOUBLE;
+				break;
+			case "char" :
+				type = VarType.CHAR;
+				break;
+			case "str" :
+				type = VarType.STRING;
+				break;
+			default :
+				throw new UnknownTypeException();
+		}
+		return type;
+	}
+	
+	private String makeString(String bTrame) {
+		String res = new String();
+		
+		for(int i=0;i<bTrame.length()-7;i+=8)
+			res += (char) Integer.parseInt(bTrame.substring(i, i+8), 2);
+		
+		return res;
+	}
+	
+	public SigfoxData decode(String trame) throws UnknownTypeException {
+		BigInteger n = new BigInteger("1" + trame, 16);
+		String bTrame = n.toString(2).substring(1);
+		SigfoxData data = new SigfoxData();
+		
+		for(int i=0;i<format.length;++i) {
+			switch(format[i].getType()) {
+			case BOOLEAN :
+				data.add(i, format[i].getName(), (Boolean) (bTrame.charAt(format[i].getBegin()) == '1'));
+				break;
+			case BYTE :
+				data.add(i, format[i].getName(), bTrame.charAt(format[i].getBegin()));
+				break;
+			case INTEGER :
+				data.add(i, format[i].getName(), Integer.parseInt(bTrame.substring(format[i].getBegin(), format[i].getEnd()+1), 2));
+				break;
+			case FLOAT :
+				data.add(i, format[i].getName(), Float.parseFloat(bTrame.substring(format[i].getBegin(), format[i].getEnd())));
+				break;
+			case DOUBLE :
+				data.add(i, format[i].getName(), Double.parseDouble(bTrame.substring(format[i].getBegin(), format[i].getEnd())));
+				break;
+			case CHAR :
+				Character c = (char) Integer.parseInt(bTrame.substring(format[i].getBegin(), format[i].getBegin()+8), 2);
+				data.add(i, format[i].getName(), c);
+				break;
+			case STRING :
+				data.add(i, format[i].getName(), makeString(bTrame.substring(format[i].getBegin(), format[i].getEnd()+1)));
+				break;
+			default :
+				throw new UnknownTypeException();
```

Should pass the unknown type in parameter `throw new UnknownTypeException(format[i].getType())